client/src/pages/ViewRequestStatus/DonationRequestStatus.js

```diff
@@ -1,12 +1,31 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import "./StatusView.css";
-import { Link } from "react-router-dom";
+import { Link, useParams } from "react-router-dom";
 import BackgroundImage from "../component/BackgroundImageComponent/BackgroundImage";
 
 function DonationRequestStatues() {
+	const [donation, setDonation] = useState("");
+	const { id } = useParams();
+
+	useEffect(() => {
+		fetch(`/api/laptop_donation/${id}`)
+			.then((res) => res.json())
+			.then((donation) => {
+				setDonation(donation);
+			});
+	}, [id]);
+	console.log(donation);
+
+	if (donation === null || donation === undefined || donation === "") {
+		return (
+			<div>
+				<BackgroundImage primaryText="Oops, seems the donation doesn't exist" />
```

Comment:
It's really good that you are able to write this kind of code quite rapidly! (just imagine how big of a task this would have been 1year ago!!)

I think it's good enough, but it means that while the donation is being fetched from the server, we might briefly see "donation doesn't exist". It would be even better to have 3 possibilities:
"" -> render nothing (initial value)
{/*non null donation*/} -> render thanks
null -> render doesn't exist

Set it to null when the status from /api/laptop_donation is 404